odoo/resource/product_product/resource.py

```
@@ -0,0 +1,106 @@
+from care.emr.resources.common.monetary_component import MonetaryComponentType
+from odoo.connector.connector import OdooConnector
+from odoo.resource.product_category.spec import CategoryData
+from odoo.resource.product_product.spec import ProductData, TaxData
+
+
+class OdooProductProductResource:
+    def get_charge_item_base_price(self, charge_item: ChargeItemDefinition):
+        for item in charge_item.price_components:
+            if item["monetary_component_type"] == MonetaryComponentType.base.value:
+                return item["amount"]
+        raise Exception("Base price not found")
```

Ruff (0.14.3)

Create your own exception

(TRY002)